### DISCUSSION TITLE
yolo detect train data=dataset.yaml model=yolo11n.pt epochs=100 imgsz=640

### DISCUSSION THREAD

**Discussion by yonglee7015** [ultralytics/ultralytics#23811] [Q&A] [answered]
yolo detect train data=dataset.yaml model=yolo11n.pt epochs=100 imgsz=640, but why does it auto download yolo26n.pt instead of  yolo11n.pt

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yonglee7015, thanks for reporting this and for using Ultralytics 🚀 This is an automated response to help collect the right details quickly, and an Ultralytics engineer will also assist you here soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many common questions are already covered.

If this is a 🐛 Bug Report, please share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can reproduce exactly what you’re seeing (your command, full console logs, `dataset.yaml`, environment details, and package version).

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by yonglee7015** [reply]
It seems that both the yolo11n.pt and yolo26n.pt files were downloaded during the process, while the model training was exclusively based on yolo26n.pt. that's why?

**Reply by glenn-jocher** [reply, marked as answer]
That switch shouldn’t happen—if you pass `model=yolo11n.pt` the trainer should load YOLO11, while a `yolo26n.pt` download can still occur as a separate cached asset; to confirm what actually got loaded, please paste the first ~30 lines of your train log (the line that echoes the resolved `model=` path and the “Ultralytics YOLO…” header) plus `yolo checks` output here, and also tell us whether the same behavior happens with the official example models from the [Models documentation](https://docs.ultralytics.com/models/).

```bash
yolo checks
yolo detect train model=yolo11n.pt data=coco8.yaml epochs=1 imgsz=640
```

**Reply by yonglee7015** [reply]
Yes, you're absolutely right. Because while it was downloading yolo26.pt, it misled me into thinking that the training should be based on this yolo26.pt file.

**Reply by glenn-jocher** [reply]
Correct — the download itself does not decide the training model; if your run starts with `model=yolo11n.pt`, then training is using YOLO11 and the `yolo26n.pt` download was just an additional asset/cache event rather than the active checkpoint.
